TGM/src/main/java/com/minehut/tgm/modules/infection/InfectionModule.java

```diff
@@ -0,0 +1,253 @@
+package com.minehut.tgm.modules.infection;
+
+import com.google.gson.JsonObject;
+import com.minehut.teamapi.models.Team;
+import com.minehut.tgm.TGM;
+import com.minehut.tgm.damage.grave.event.PlayerDeathByPlayerEvent;
+import com.minehut.tgm.damage.grave.event.PlayerDeathEvent;
+import com.minehut.tgm.damage.grave.listener.PlayerListener;
+import com.minehut.tgm.damage.tracker.event.PlayerDamageEvent;
+import com.minehut.tgm.match.Match;
+import com.minehut.tgm.match.MatchModule;
+import com.minehut.tgm.match.MatchStatus;
+import com.minehut.tgm.modules.countdown.BossBarCountdown;
+import com.minehut.tgm.modules.countdown.Countdown;
+import com.minehut.tgm.modules.scoreboard.ScoreboardManagerModule;
+import com.minehut.tgm.modules.scoreboard.SimpleScoreboard;
+import com.minehut.tgm.modules.team.MatchTeam;
+import com.minehut.tgm.modules.team.TeamChangeEvent;
+import com.minehut.tgm.modules.team.TeamManagerModule;
+import com.minehut.tgm.user.PlayerContext;
+import com.minehut.tgm.util.Players;
+import lombok.Getter;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.GameMode;
+import org.bukkit.boss.BarColor;
+import org.bukkit.boss.BarStyle;
+import org.bukkit.boss.BossBar;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.potion.Potion;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.scoreboard.NameTagVisibility;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Random;
+
+/**
+ * Created by Draem on 7/31/2017.
+ */
+public class InfectionModule extends MatchModule implements Listener {
+
+    @Getter private TeamManagerModule teamManager;
+
+    @Getter private int length;
+
+    @Getter private Match match;
+
+    @Override
+    public void load(Match match) {
+        TGM.registerEvents(this);
+        JsonObject json = match.getMapContainer().getMapInfo().getJsonObject().get("infection").getAsJsonObject();
+        length = json.get("length").getAsInt();
```

Comment:
This should not be a required field. We should be able to create maps that don't have a time limit (done by not including a "length" field in the map.json)